src/Console/Util/PhpProcess.php

```diff
@@ -9,85 +9,31 @@
 
 namespace Infection\Console\Util;
 
-use Composer\XdebugHandler\XdebugHandler;
+use Composer\XdebugHandler\PhpConfig;
 use Symfony\Component\Process\Process;
 
 /**
- * Vanilla PHP process and utility functions
- *
  * @internal
  */
 final class PhpProcess extends Process
 {
     /**
-     * @var string|bool
-     */
-    private static $phprc;
-
-    /**
-     * Setups a default Xdebug-free environment for all subprocesses:
+     * Runs a PHP process with xdebug loaded
+     *
+     * If xdebug was loaded in the main process, it will have been restarted
+     * without xdebug and configured to keep xdebug out of PHP sub-processes.
      *
-     * - PHPRC should point to our temporary php.ini (we need to save a previous value)
+     * This method allows a sub-process to run with xdebug enabled (if it was
+     * originally loaded), then restores the xdebug-free environment.
      *
-     * - PHP_INI_SCAN_DIR should be made blank because our php.ini has all we need
-     *   (a previous value can be found in XdebugHandler::getRestartSettings()["scanDir"])
+     * {@inheritdoc}
      */
-    public static function setupXdebugFreeEnvironment(): void
-    {
-        if (!isset(self::$phprc)) {
-            self::$phprc = getenv('PHPRC');
-        }
-
-        self::putenv('PHPRC', XdebugHandler::getRestartSettings()['tmpIni']);
-        self::putenv('PHP_INI_SCAN_DIR', '');
-    }
-
     public function start(callable $callback = null, array $env = null): void
     {
-        // Xdebug wasn't skipped, running as is
-        if ('' === XdebugHandler::getSkippedVersion()) {
-            parent::start($callback, $env ?? []);
-
-            return;
-        }
-
-        /*
-         * Vanilla processes are expected to run in a vanilla, or the most original, environment.
-         *
-         * For that, we need to remove or reset all environment variables we set to setup our custom
-         * xdebug-free environment, later setting them back so we won't have to think again about it
-         * for the bulk of other processes, which not require xdebug to function, and work better
-         * without it.
-         */
-
-        self::restoreVanillaEnvironment();
+        $phpConfig = new PhpConfig();
```

Comment:
@sanmai Was there a specific reason that we did not use the `PhpConfig` class before?

Reply:
@BackEndTea yes. It didn't exist :) 

Was introduced in the 1.2 https://github.com/composer/xdebug-handler/releases

Reply:
@sanmai was waiting for me to release it, but I got a bit behind.

Reply:
Well then that makes sense :smile: :